### ISSUE TITLE
Disable Corruption/Crimson/Hallow spread not working

### ISSUE THREAD

**Issue by PlutiaHeart** [NyxStudios/TShock#1244]
I'm new to this, I made my account just for this issue, in fact, so if anything's wrong with the post please let me know

I moved my server onto tshock primarily to take advantage of the option of disabling biome spread... only to find that when I went to a partially converted hallowed desert, it was still spreading. I looked into the code to see if it was something simple... 

			if (!Config.AllowCrimsonCreep && (args.Type == TileID.Dirt || args.Type == TileID.FleshWeeds
				|| TileID.Sets.Crimson[args.Type]))
			{
				args.Handled = true;
				return;
			}

			if (!Config.AllowCorruptionCreep && (args.Type == TileID.Dirt || args.Type == TileID.CorruptThorns
				|| TileID.Sets.Corrupt[args.Type]))
			{
				args.Handled = true;
				return;
			}

			if (!Config.AllowHallowCreep && (TileID.Sets.Hallow[args.Type]))
			{
				args.Handled = true;
}

Now, I'm not even going to pretend I'm an expert in this, but from what I can tell, it's only checking if it spreads to dirt tiles for crimson and corruption, and for Hallow, it's not checking anything at all?
Could someone please explain how this works?

**Comment by WhiteXZ** [comment]
`TileID.Sets.Corrupt`, `TileID.Sets.Hallow`, and `TileID.Sets.Crimson` contain the other tile IDs that are checked

**Comment by PlutiaHeart** [comment]
Thanks for the clarification.

**Comment by WhiteXZ** [comment]
Upon review, it appears that those sets may have been missing a few biome spreading tiles. https://github.com/NyxStudios/TerrariaAPI-Server/commit/0c3280f9ff0fde3d73e57a6ecff2f206a34a976f should hopefully resolve that

**Comment by Patrikkk** [comment]
This should be fixed by now in latest release, no?

**Comment by Kolatra** [comment]
Still not fixed.

**Comment by daredev** [comment]
@WhiteXZ @ProfessorXZ has [this](https://github.com/NyxStudios/TShock/issues/1244#issuecomment-233499627) (commit: 0c3280f9ff0fde3d73e57a6ecff2f206a34a976f) been merged into one of the releases or do we need to get the unstable build to test if it solved the problem?

**Comment by ProfessorXZ** [comment]
It was merged in v4.3.18

**Comment by WhiteXZ** [comment]
It's possible that the relevant hooks were not re-added

**Comment by nicatronTg** [comment]
Adding PR wanted tag though this is really more api

**Comment by Patrikkk** [comment]
I've just looked into this.  The invokes are intact in the WorldGen classs. I placed breakpoints at the invokes, and also inside TShock where it handles the event if conditions are correct. 
It returned from the spreading.  So it seems all good to me. 
I've asked around on my server and noone noticed any spreading, so I guess we can close this issue. 
It *should* be functioning properly. 
If anyone else notices any spreading, they can comment here  with detailed information to have the issue re-opened.
